sklearn/tests/test_common.py

```diff
@@ -143,6 +144,26 @@ def test_root_import_all_completeness():
         assert_in(modname, sklearn.__all__)
 
 
+def test_all_tests_are_importable():
+    # Ensure that for each contentful subpackage, there is a test directory
+    # within it that is also a subpackage (i.e. a directory with __init__.py)
+
+    HAS_TESTS_EXCEPTIONS = re.compile(r'''(?x)
+                                      \.externals(\.|$)|
+                                      \.tests(\.|$)|
+                                      \._
+                                      ''')
+    lookup = dict((name, ispkg)
+                  for _, name, ispkg
+                  in pkgutil.walk_packages(sklearn.__path__,
+                                           prefix='sklearn.'))
+    missing_tests = [name for name, ispkg in lookup.items()
+                     if ispkg
+                     and not HAS_TESTS_EXCEPTIONS.search(name)
+                     and name + '.tests' not in lookup]
+    assert_false(missing_tests)
```

Comment:
It will... It says the likes of `['sklearn.utils.sparsetools.tests'] is not False` which is not as explicit as it could be, but ...